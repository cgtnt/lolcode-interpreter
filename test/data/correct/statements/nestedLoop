HAI

IM IN YR I HAS A COUNT ITZ 1, WILE DIFFRINT COUNT AN BIGGR OF COUNT AN 5 
  IM IN YR I HAS A COUNTWO ITZ 1, WILE DIFFRINT COUNTWO AN BIGGR OF COUNTWO AN 5 
    VISIBLE SMOOSH "COUNT IZ " AN COUNT AN " " AN COUNTWO MKAY!
    COUNTWO R SUM OF COUNTWO AN 1
  IM OUTTA YR
  COUNT R SUM OF COUNT AN 1
IM OUTTA YR

VISIBLE "DONE!"

KTHXBYE
