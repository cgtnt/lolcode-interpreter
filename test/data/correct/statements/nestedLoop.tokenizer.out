@1:T_BEGIN:HAI-@3:T_COMMAND_TERMINATOR:
-@3:T_LOOP_BEGIN:IM IN YR-@3:T_DECLARE_VAR:I HAS A-@3:T_T_IDENTIFIER:COUNT-@3:T_DECLARE_SET_VAR:ITZ-@3:T_T_INT:1-@3:T_COMMAND_TERMINATOR:,-@3:T_WHILE:WILE-@3:T_NOT_EQUAL:DIFFRINT-@3:T_T_IDENTIFIER:COUNT-@3:T_AND:AN-@3:T_MAX:BIGGR OF-@3:T_T_IDENTIFIER:COUNT-@3:T_AND:AN-@3:T_T_INT:5-@4:T_COMMAND_TERMINATOR:
-@4:T_LOOP_BEGIN:IM IN YR-@4:T_DECLARE_VAR:I HAS A-@4:T_T_IDENTIFIER:COUNTWO-@4:T_DECLARE_SET_VAR:ITZ-@4:T_T_INT:1-@4:T_COMMAND_TERMINATOR:,-@4:T_WHILE:WILE-@4:T_NOT_EQUAL:DIFFRINT-@4:T_T_IDENTIFIER:COUNTWO-@4:T_AND:AN-@4:T_MAX:BIGGR OF-@4:T_T_IDENTIFIER:COUNTWO-@4:T_AND:AN-@4:T_T_INT:5-@5:T_COMMAND_TERMINATOR:
-@5:T_WRITE_STDOUT:VISIBLE-@5:T_CONCAT:SMOOSH-@5:T_T_STRING:"COUNT IZ "-@5:T_AND:AN-@5:T_T_IDENTIFIER:COUNT-@5:T_AND:AN-@5:T_T_STRING:" "-@5:T_AND:AN-@5:T_T_IDENTIFIER:COUNTWO-@5:T_END_INF:MKAY-@5:T_BANG:!-@6:T_COMMAND_TERMINATOR:
-@6:T_T_IDENTIFIER:COUNTWO-@6:T_ASSIGN:R-@6:T_PLUS:SUM OF-@6:T_T_IDENTIFIER:COUNTWO-@6:T_AND:AN-@6:T_T_INT:1-@7:T_COMMAND_TERMINATOR:
-@7:T_LOOP_END:IM OUTTA YR-@8:T_COMMAND_TERMINATOR:
-@8:T_T_IDENTIFIER:COUNT-@8:T_ASSIGN:R-@8:T_PLUS:SUM OF-@8:T_T_IDENTIFIER:COUNT-@8:T_AND:AN-@8:T_T_INT:1-@9:T_COMMAND_TERMINATOR:
-@9:T_LOOP_END:IM OUTTA YR-@11:T_COMMAND_TERMINATOR:
-@11:T_WRITE_STDOUT:VISIBLE-@11:T_T_STRING:"DONE!"-@13:T_COMMAND_TERMINATOR:
-@13:T_END:KTHXBYE-@14:T_COMMAND_TERMINATOR:
-@14:T_EOF:
